protocols/gossipsub/src/types.rs

```diff
@@ -72,11 +72,15 @@ macro_rules! declare_message_id_type {
 // A type for gossipsub message ids.
 declare_message_id_type!(MessageId, "MessageId");
 
+// mxinden: Do I understand correctly that a `FastMessageId` is used to reduce cost of hashing given
```

Comment:
We probably should rename this to duplicate_message_id. 

Its purpose is to avoid doing decompression when compression is used. Depending on the compression and size of messages, there is significant difference between taking a hash and decompressing a message and taking a hash.